### PR TITLE
Fix "fast_training_tutorial.ipynb"

### PR DESCRIPTION
**Pull request by KumoLiu**
Fixes https://github.com/Project-MONAI/tutorials/issues/1250.
Fixes https://github.com/Project-MONAI/MONAI/issues/6149

### Description
Now when `Resize` accept the same spatial_size with the original shape and `set_track_meta(False)`, it will only return meta information.
https://github.com/Project-MONAI/MONAI/blob/9fd6d4ced3efb344cf3b24c51d899d8fafc882d5/monai/transforms/spatial/functional.py#L297-L301

In fast_training_tutorial.ipynb, when in "fast" mode, it will `set_track_meta(False)` which causes error when doing `Resize`
This pr change `sliding_window_inference` to only `Resize` when the shape is different.
https://github.com/Project-MONAI/MONAI/blob/a2ec3752f54bfc3b40e7952234fbeb5452ed63e3/monai/inferers/utils.py#L253

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/inferers/utils.py [hunk 1]
```diff
@@ -255,7 +255,11 @@ def sliding_window_inference(
                             "Tips: if overlap*roi_size*zoom_scale is an integer, it usually works."
                         )
                     original_idx_zoom[axis] = slice(int(zoomed_start), int(zoomed_end), None)
-                importance_map_zoom = resizer(importance_map.unsqueeze(0))[0].to(compute_dtype)
+                importance_map_zoom = (
+                    resizer(importance_map.unsqueeze(0))[0].to(compute_dtype)
```
Comment (wyli) [inline monai/inferers/utils.py:259]: thanks for looking into this issue, for the resize functional, I think we should also fix the logic when orig_size==out_size? (this is untested):
```py
    if transform_info.get(TraceKeys.LAZY_EVALUATION, False):
        if anti_aliasing and transform_info.get(TraceKeys.LAZY_EVALUATION, False):
            warnings.warn("anti-aliasing is not compatible with lazy evaluation.")
        return out.copy_meta_from(meta_info) if isinstance(out, MetaTensor) else meta_info
    if tuple(convert_to_numpy(orig_size)) == out_size:
        return out.copy_meta_from(meta_info) if isinstance(out, MetaTensor) else out
```

Comment (KumoLiu) [reply monai/inferers/utils.py:259]: Have changed the logic in `Resize` and added a unit test for `set_track_meta(False)`.

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me. I'll include minor integration tests https://github.com/Project-MONAI/MONAI/compare/dev...6149-integraion

File: monai/transforms/spatial/functional.py [hunk 2]
```diff
@@ -295,10 +295,12 @@ def resize(img, out_size, mode, align_corners, dtype, input_ndim, anti_aliasing,
         lazy_evaluation=transform_info.get(TraceKeys.LAZY_EVALUATION, False),
     )
     out = convert_to_tensor(img.as_tensor() if isinstance(img, MetaTensor) else img, track_meta=get_track_meta())
-    if transform_info.get(TraceKeys.LAZY_EVALUATION, False) or tuple(convert_to_numpy(orig_size)) == out_size:
+    if transform_info.get(TraceKeys.LAZY_EVALUATION, False):
         if anti_aliasing and transform_info.get(TraceKeys.LAZY_EVALUATION, False):
             warnings.warn("anti-aliasing is not compatible with lazy evaluation.")
         return out.copy_meta_from(meta_info) if isinstance(out, MetaTensor) else meta_info
+    if tuple(convert_to_numpy(orig_size)) == out_size:
+        return out.copy_meta_from(meta_info) if isinstance(out, MetaTensor) else out
```
Comment (wyli) [inline monai/transforms/spatial/functional.py:303]: to be consistent with line 321, this should always return float32...

also there's the same issue here https://github.com/Project-MONAI/MONAI/blob/9fd6d4ced3efb344cf3b24c51d899d8fafc882d5/monai/transforms/spatial/functional.py#L135-L138

I'll update

**Pull request merged**